src/features/charts/redux/middleware.ts

```diff
@@ -0,0 +1,15 @@
+import { Middleware, AnyAction } from 'redux'
+import { UPDATE_CHART_RANGE } from '../redux'
+
+export const chartMiddleware = (socket: SocketIOClient.Socket): Middleware => {
+    return ({getState}) => {
+        return (next) => (action: AnyAction) => {
```

Comment:
`AnyAction` is a redundant type declaration, as you've already typed the whole thing as a `Middleware`

Reply:
This was implemented before Middleware was typed, I've gone back and removed them